File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int carry=0;
        ArrayList<Integer> ans = new ArrayList<>();
        ListNode one = l1;
        ListNode two = l2;
        while(one!=null && two!=null){
        int sum = (one.val + two.val + carry)%10;
        ans.add(sum);
         carry = (one.val + two.val + carry) / 10;
         one = one.next;
         two = two.next;
        }
        //if(carry>0) ans.add(carry);
        while(one!=null){
            int temp = (one.val + carry)%10;
            ans.add(temp);
            carry = (one.val + carry) / 10;
            one = one.next;
        }
        while(two!=null){
            int temp = (two.val + carry)%10;
            ans.add(temp);
            carry = (two.val + carry) / 10;
            two = two.next;
        }
        if(carry>0) ans.add(carry);
        ListNode dummy = new ListNode(0);
        ListNode dummy1 = dummy;
        for(int v : ans){
            dummy1.next = new ListNode(v);
            dummy1 = dummy1.next;
        }
        return dummy.next;
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        one = l1
        two = l2
        ans = []
        carry = 0
        while one!=None and two!=None:
            sum = (one.val+two.val+carry)%10
            ans.append(sum)
            carry = (one.val+two.val+carry)//10
           
            one=one.next
            two=two.next
        while one!=None:
            sum = (one.val+carry)%10
            ans.append(sum)
            carry = (one.val+carry)//10
            
            one = one.next
        while two!=None:
            sum = (two.val+carry)%10
            ans.append(sum)
            carry = (two.val+carry)//10
           
            two = two.next
        if carry>0:
            ans.append(carry)
        
        dummy = ListNode(0)
        dummy1 = dummy
        for num in ans:
            dummy1.next = ListNode(num)
            dummy1 = dummy1.next
        return dummy.next
            

        